app.py

```diff
@@ -265,344 +230,386 @@ def create_donut_chart_column(figure_id):
+    value_counts = cursor.fetchall()  # List of tuples
+    cursor.close()
+    mariadb_connection.close()
+
+    # Convert list of tuples to dictionary
+    reload_value_counts = dict(value_counts)
+
+    # Remap byte keys to string values
+    key_mapping = {b"\x00": "New Game", b"\x01": "Reloaded Game"}
+    reload_value_counts = {
+        key_mapping[key]: value for key, value in reload_value_counts.items()
+    }
+
     figure = px.pie(
-        reload_value_counts,
-        names=reload_value_counts.index,
-        values=reload_value_counts['count'],
-        title='Reloaded Games',
+        names=list(reload_value_counts.keys()),
+        values=list(reload_value_counts.values()),
+        title="Reloaded Games",
         hole=0.7,
     )
     figure.update_layout(
         hoverlabel=dict(
             bgcolor="white",
         )
     )
+    figure.update_traces(hovertemplate="<b>%{label}</b><br>%{value:,}")
     return figure
 
 
 @callback(
-    Output('observers-chart', 'figure'),
-    Input('table', 'data'),
-    State('table', 'columns'),
+    Output("observers-chart", "figure"),
+    Input("date-picker", "start_date"),
+    Input("date-picker", "end_date"),
 )
-def update_observers_chart(data, columns):
+def update_observers_chart(start_date, end_date):
     """
-    Update the observers-chart whenever the table data changes.
+    Update the observers-chart whenever the date range changes.
 
     Args:
-        data (list[dict]): The data to update the chart with.
-        columns (list[dict]): The columns of the data.
+        start_date (str): The start date of the selected date range.
+        end_date (str): The end date of the selected date range.
 
     Returns:
         plotly.graph_objects.Figure: The updated observers-chart.
     """
-    df = pd.DataFrame(data, columns=[column['name'] for column in columns])
-    observers_value_counts = (
-        df['OBSERVERS']
-        .replace({
-            0: 'Observers not allowed',
-            1: 'Observers allowed'
-        })
-        .value_counts()
-        .to_frame()
+    # Validate that both start_date and end_date are not None.
+    if start_date is None or end_date is None:
+        return px.pie()  # An empty chart
+
+    # Query the database.
+    mariadb_connection = connect_to_mariadb()
+    cursor = mariadb_connection.cursor()
+    target_table = "wesnothd_game_info"
+    cursor.execute(
+        f"SELECT OBSERVERS, COUNT(*) FROM {target_table} WHERE START_TIME BETWEEN ? AND ? GROUP BY OBSERVERS",
+        (start_date, end_date),
     )
+    logging.debug(f"Fetched OBSERVERS value counts of {target_table} from database")
+    value_counts = cursor.fetchall()  # List of tuples
+    cursor.close()
+    mariadb_connection.close()
+
+    # Convert list of tuples to dictionary
+    observers_value_counts = dict(value_counts)
+
+    # Remap byte keys to string values
+    key_mapping = {b"\x00": "Observers allowed", b"\x01": "Observers not allowed"}
```

Comment:
The mapping got switched here. 1 is observers allowed.

Reply:
Thank you for spotting this. Fixed now in 99b4966